adapters/rack/lib/opentelemetry/adapters/rack/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,253 @@
+                                       'component' => 'http',
+                                       'service' => config[:web_service_name],
+                                       'start_time' => request_start_time
+                                     },
+                                     kind: :server)
+
+            extracted_context.set_value(current_span_key, span)
+          end
+
+          def finish_span(context)
+            context[current_span_key]&.finish if context
+          end
+
+          def current_span_key
+            OpenTelemetry::Trace::Propagation::ContextKeys.current_span_key
+          end
+
+          def tracer
+            OpenTelemetry::Adapters::Rack::Adapter.instance.tracer
+          end
+
+          ### request_span
+
+          # Per https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#http-server-semantic-conventions
+          #
+          # One of the following sets is required, in descending preference:
+          # * http.scheme, http.host, http.target
+          # * http.scheme, http.server_name, net.host.port, http.target
+          # * http.scheme, net.host.name, net.host.port, http.target
+          # * http.url
+          def request_span_attributes(env:)
+            {
+              'component' => 'http',
+              'http.method' => env['REQUEST_METHOD'],
+              'http.host' => env['HOST'],
+              'http.scheme' => env['rack.url_scheme'],
+              'http.target' => fullpath(env)
+            }.merge(allowed_request_headers(env))
+          end
+
+          # e.g., "/webshop/articles/4?s=1":
+          def fullpath(env)
+            query_string = env['QUERY_STRING']
+            path = env['SCRIPT_NAME'] + env['PATH_INFO']
+
+            query_string.empty? ? path : "#{path}?#{query_string}"
+          end
+
+          # https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#name
+          #
+          # recommendation: span.name(s) should be low-cardinality (e.g.,
+          # strip off query param value, keep param name)
+          #
+          # see http://github.com/open-telemetry/opentelemetry-specification/pull/416/files
+          def create_request_span_name(request_uri_or_path_info)
+            # NOTE: dd-trace-rb has implemented 'quantization' (which lowers url cardinality)
+            #       see Datadog::Quantization::HTTP.url
+
+            if (implementation = config[:url_quantization])
+              implementation.call(request_uri_or_path_info)
+            else
+              request_uri_or_path_info
+            end
+          end
+
+          # catch exceptions that may be raised in the middleware chain
+          # Note: if a middleware catches an Exception without re raising,
+          # the Exception cannot be recorded here.
+          def record_and_reraise_error(error, request_span:)
+            request_span.record_error(error)
```

Comment:
`Span#record_error` sets the status for us now. If you use the alternate call (that uses `in_span`) we won't need this method. If we keep it, it doesn't need to set the status.